lti_consumer/plugin/compat.py

```diff
@@ -19,3 +21,47 @@ def run_xblock_handler_noauth(*args, **kwargs):
     # pylint: disable=import-error,import-outside-toplevel
     from lms.djangoapps.courseware.module_render import handle_xblock_callback_noauth
     return handle_xblock_callback_noauth(*args, **kwargs)
+
+
+def get_user_from_external_user_id(external_user_id):
+    """
+    Import ExternalId model and find user by external_user_id
+    """
+    # pylint: disable=import-error,import-outside-toplevel
+    from openedx.core.djangoapps.external_user_ids.models import ExternalId
+    try:
+        external_id = ExternalId.objects.get(
+            external_user_id=external_user_id,
+            external_id_type__name='lti'
+        )
+        return external_id.user
+    except ExternalId.DoesNotExist as exception:
+        raise LtiError('Invalid User') from exception
+    except ValidationError as exception:
+        raise LtiError('Invalid userID') from exception
+
+
+def submit_grade(score):
+    """
+    Import grades signals and submit grade given a LtiAgsScore instance.
+    """
+    # pylint: disable=import-error,import-outside-toplevel
+    from lms.djangoapps.grades.api import signals as grades_signals
+
+    # get lti config
+    lti_config = score.line_item.lti_configuration
+
+    # find user
+    user = get_user_from_external_user_id(score.user_id)
+
+    # publish score
+    grades_signals.SCORE_PUBLISHED.send(
+        sender=None,
+        block=lti_config.block,
```

Comment:
You should use the block linked by `resource_link_id`, not the one from `lti_config`. 
Though both are the same now, it's not how they will be in the future.

Reply:
@giovannicimolin, Now it uses the block from LineItem. And the block is loaded using ``load_block_as_anonymous_user``.